utils/multi_frameworks_dataset.py

```diff
@@ -19,8 +30,32 @@ class MultiFrameworkDataset(BaseDataset, ABC):
         'framework': ['pytorch', 'lightning', 'tensorflow'],
     }
 
-    install_cmd = 'conda'
-    requirements = ['pip:tensorflow-datasets']
+    install_cmd = "conda"
+    requirements = ["pip:tensorflow-datasets", "scikit-learn"]
+
+    def get_registration_indices(self):
+        registration_dir = Path("./torch_tf_datasets_registrations/")
+        filepath = registration_dir / f"{self.tf_ds_name}_train.npy"
+        if filepath.exists():
+            return np.load(filepath)
+        print('Registration file not found')
+        return None
+
+    def set_train_val_indices(self):
+        registration_indices = self.get_registration_indices()
+        if registration_indices is None:
+            self.train_val_split_spec = False
+        self.tf_train_indices, self.tf_val_indices = train_test_split(
+            np.arange(len(registration_indices)),
+            test_size=self.ds_description["n_samples_val"],
+            train_size=self.ds_description["n_samples_train"],
+            random_state=self.random_state,
+        )
+        self.tf_train_indices = np.sort(self.tf_train_indices)
+        self.tf_val_indices = np.sort(self.tf_val_indices)
+        self.torch_train_indices = registration_indices[self.tf_train_indices]
+        self.torch_val_indices = registration_indices[self.tf_val_indices]
+        self.train_val_split_spec = True
```

Comment:
Not a big fan of storing all this and pass it in an implicit manners.
I would do something like:
```suggestion
        registration_indices = self.get_registration_indices()

        tf_train_indices, tf_val_indices = train_test_split(
            np.arange(len(registration_indices)),
            test_size=self.ds_description["n_samples_val"],
            train_size=self.ds_description["n_samples_train"],
            random_state=self.random_state,
        )
        tf_train_indices.sort(), tf_val_indices.sort()
        if self.framework == "tensorflow" or registration_indices is None:
            return tf_train_indices, tf_val_indices

        torch_train_indices = registration_indices[tf_train_indices]
        torch_val_indices = registration_indices[tf_val_indices]
        return torch_train_indices, torch_val_indices
```